controllers/upgrade_handlers.go

```diff
@@ -18,39 +18,104 @@ package controllers
 
 import (
 	"context"
+	"errors"
 	"fmt"
+	"strings"
+	"time"
+
+	velerov1 "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
 	lcav1alpha1 "github.com/openshift-kni/lifecycle-agent/api/v1alpha1"
 	"github.com/openshift-kni/lifecycle-agent/controllers/utils"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"github.com/openshift-kni/lifecycle-agent/internal/backuprestore"
 	ctrl "sigs.k8s.io/controller-runtime"
 )
 
+// custom error for the cases when we would need to reject or accept requeue. Msg is expected to be updated with more info when in use
+var (
+	BackupFailed                       = errors.New("backup failed, will not requeue")
+	BackupValidationFailedShortRequeue = errors.New("backup validation failed, please update")
+)
+
+// handleUpgrade orchestrate main upgrade steps
 func (r *ImageBasedUpgradeReconciler) handleUpgrade(ctx context.Context, ibu *lcav1alpha1.ImageBasedUpgrade) (ctrl.Result, error) {
+	r.Log.Info("Starting handleUpgrade")
+
+	// backup with OADP
+	r.Log.Info("Backup with Oadp operator")
+	reqBackup, backupCRs, err := r.BackupRestore.CheckIfBackupRequested(ibu.Spec.OADPContent, ctx)
+	if err != nil {
+		if k8serrors.IsInvalid(err) {
+			utils.SetStatusCondition(&ibu.Status.Conditions,
+				utils.GetInProgressConditionType(lcav1alpha1.Stages.Upgrade),
+				utils.ConditionReasons.Failed,
+				metav1.ConditionTrue,
+				fmt.Sprintf("Invalid backup resource detected in configMap, please updated"),
+				ibu.Generation)
+			return requeueWithShortInterval(), nil
+		}
+		return requeueWithError(err)
+	}
+	if reqBackup {
+		r.Log.Info("Backup requested")
+		ctrlResult, err := r.handleBackup(ctx, backupCRs)
+		if err != nil {
+			switch {
+			case errors.Is(err, BackupFailed):
+				utils.SetStatusCondition(&ibu.Status.Conditions,
+					utils.GetInProgressConditionType(lcav1alpha1.Stages.Upgrade),
+					utils.ConditionReasons.Failed,
+					metav1.ConditionFalse,
```

Comment:
The status should be True. And we also need to update the UpgradeCompleted condition.

Reply:
Please ignore this comment. The status should be reflected to the condition type so it should be False